### PR TITLE
fix(rpc middleware): fix `is_error` bug

### PR DESCRIPTION
**Pull request by niklasad1**
Closing #11887

### CODE REVIEW

**Review by lexnv — APPROVED** [review]

File: client/rpc-servers/src/middleware.rs [hunk 1]
```diff
@@ -176,7 +176,9 @@ impl Middleware for RpcMiddleware {
 			.with_label_values(&[
 				self.transport_label,
 				name,
-				if success { "true" } else { "false" },
+				// the label "is_error", so `success` should be regarded false
+				// and vice-versa to be registrered correctly.
+				if success { "false" } else { "true" },
```
Comment (breathx) [inline client/rpc-servers/src/middleware.rs:181]:
```suggestion
				!success,
```

Comment (ggwpez) [reply client/rpc-servers/src/middleware.rs:181]: It expects a string, not a bool.

File: client/rpc-servers/src/middleware.rs [hunk 2]
```diff
@@ -176,7 +176,9 @@ impl Middleware for RpcMiddleware {
 			.with_label_values(&[
 				self.transport_label,
 				name,
-				if success { "true" } else { "false" },
+				// the label "is_error", so `success` should be regarded false
```
Comment (niklasad1) [inline client/rpc-servers/src/middleware.rs:179]:
```suggestion
				// the label "is_error", so `success` should be regarded as false
```

**Review by bkchr — APPROVED** [review]

**Comment by niklasad1** [comment]
bot merge

**Pull request merged**